docs/updating-the-docs.md

```diff
@@ -48,3 +46,26 @@ Examples are included in the markdown files using the [pymdown snippet notation]
 Make sure you commit both the changes to the `examples` and `databuilder/snippets` directories.
 
 If you are updating multiple examples it might be easier to the extraction step happen each time you change the python modules, you can do this with `just databuilder/watch-examples`.
+
+
+## Updating Data Builder Backend and Contract documentation
+
+!!! note "You will need both Python 3.7 and 3.10 on your PATH to make these updates"
+
+Some of the Data Builder documentation is generated from parts of databuilder's codebase.
+
+There are three components used to achieve this:
+
+* [Data Builder](https://github.com/opensafely-core/databuilder/)
+* [A mkdocs plugin](https://github.com/opensafely-core/mkdocs-opensafely-databuilder/)
+* [This documentation site](https://github.com/opensafely/documentation/)
+
+To update the [Backend](data-backends.md) or [Contracts](contracts-reference.md) pages:
+
+1. Make your changes in the [Data Builder repo](https://github.com/opensafely-core/databuilder/).  A new version of databuilder will be released once your changes are merged to `main`.
+1. In the [documentation repo](https://github.com/opensafely/documentation/) run `just databuilder/upgrade-databuilder` to update the version of databuilder installed in the subdirectory's virtualenv.
```

Comment:
I think we should emphasize that this is done after a new version of databuilder is released. (Do all merges to `main` prompt a new release?)

Reply:
> Do all merges to main prompt a new release?

Yep!  The `build_and_deploy` workflow tags the repo based on the semantic commit used, and builds a new docker image.  Technically there's a race condition where you could hit merge, and then run `just databuilder/upgrade-databuilder` before the tagging flow finishes, but I suspect it's very unlikely to happen and I'm not sure it would be a valuable addition to the docs?

Reply:
I think I've misunderstood. The `main` workflow's `tag-new-version` step doesn't always create a new release (only for `fix`, `perf`, `feat`, and `BREAKING CHANGE`). If I merged a `docs` change to `main`, such as changing the docstring for `PatientDemographics`, then a new release wouldn't be created, so this step would be a noop, wouldn't it?

Reply:
TIL `docs` doesn't create a new release… arg.

I'll amend to make this part conditional, thanks for catching this!

Reply:
Thank you 🙂

Reply:
`github-tag-action` never fails to be confusing.

Reply:
Oh wait, we actually need a new version of databuilder to be built so we can use the tooling here because we only install a versioned build of it 😩

The alternative is to run the command with a local copy of the databuilder and then copy `public_docs.json` over, but I think that's going to make the documentation here very complicated.  However, the alternative feels like a foot gun waiting to happen 🤔

Reply:
comment: I think this still need addressing :arrow_up:

Reply:
suggestion: The quick fix might just be to say:
* a current limitation is that you need to wait for a new release before rebuilding the generated content
* note that releases are only made for certain pull requests